pkg/server/bundle/client/client.go

```diff
@@ -72,6 +73,12 @@ func NewClient(config ClientConfig) (Client, error) {
 func (c *client) FetchBundle(ctx context.Context) (*bundleutil.Bundle, error) {
 	resp, err := c.client.Get(c.c.EndpointURL)
 	if err != nil {
+		var hostnameError *x509.HostnameError
```

Comment:
I was doing some quick testing and noticed that the x509 stack returns this error as a struct value (i.e. `x509.HostnameError{}`), not a pointer to struct value (i.e. `&x509.HostnameError`). This means that errors.As will return false unless passed a pointer to the struct, not a pointer to a pointer.
```suggestion
		var hostnameError x509.HostnameError
```

I can push a commit with my tests.

Reply:
Fix: https://github.com/spiffe/spire/pull/3395/commits/9562435bafa55c288b401aeb3cfe0265bf270d6e